tiledb/cloud/dag/dag.py

```diff
@@ -120,37 +133,60 @@ def finished(self):
 
     done = finished
 
-    def exec(self, namespace=None):
+    def exec(self, namespace: Optional[str] = None):
         """
         Execute function for node
         :return: None
         """
-        self.status = st.Status.RUNNING
-
+        assert self.dag
         # Override namespace if passed
         if namespace is not None and not self.local_mode:
             self.kwargs["namespace"] = namespace
-
-        # First loop though any non-default parameter arguments to find any nodes
-        # If there is a node as an argument, the user really just wants the results, so let's fetch them
-        # and swap out the parameter
-        if self.args is not None:
-            self.args = _replace_nodes_with_results(self.args)
-
-        # First loop though any default named parameter arguments to find any nodes
-        # If there is a node as an argument, the user really just wants the results, so let's fetch them
-        # and swap out the parameter
-        if self.kwargs is not None:
-            self.kwargs = _replace_nodes_with_results(self.kwargs)
-
-        # Execute user function with the parameters that the user requested
-        # The function is executed on the dag's worker pool
-        self.future = self.dag.executor.submit(self.func, *self.args, **self.kwargs)
+        self.status = st.Status.RUNNING
+        self.future = self.dag.executor.submit(self._do_exec)
+        assert self.future  # Needed to silence mypy.
         self.future.add_done_callback(self._handle_completed_future)
 
+    def _do_exec(self) -> _T:
+        if self._is_remote:
+            # Stored parameters work only on remote functions.
+            (args, kwargs), seen = _replace_nodes_with_stored_params(
+                (self.args, dict(self.kwargs))
+            )
+            # TODO: We can't use stored params in SQL queries.
```

Comment:
(making sure there is a story for this, especially if it might break some existing use-case?)